### ISSUE TITLE
[BUG]: DateOnly / TimeOnly deserialization fails due to validation errors

### ISSUE THREAD

**Issue by Mitkee** [aloneguid/parquet-dotnet#395]
### Library Version

4.15.0 - 4.16.3

### OS

Windows, however likely a general issue

### OS Architecture

64 bit

### How to reproduce?

Have a class / record with a DateOnly or TimeOnly property, ie 

```
       record RecordContainingDateAndtimeOnly {
            public DateOnly? NullableDate { get; set; }
            public TimeOnly? NullableTime { get; set; }
            public DateOnly Date { get; set; }
            public TimeOnly Time { get; set; }
        }
```
Call serialize, then deserialize the result. 

The date only & time only fields will correcty serialize and be stored. However, upon reading the serialized values, we get a type mismatch in between the schema and the expected type: 

`System.IO.InvalidDataException
property 'NullableDate' is declared as 'NullableDate (System.DateOnly?)' but source data has it as 'NullableDate (System.DateTime?)'`

Note that if commenting out the added validation in ParquetSerializer.cs:190 
```
                if(actual != null && !actual.IsArray && !fasm.Field.Equals(actual)) {
                    throw new InvalidDataException($"property '{fasm.Field.ClrPropName}' is declared as '{fasm.Field}' but source data has it as '{actual}'");
                }

```

..then the test (supplied below)  passes and the types are deserialized as expected. 

Likely the special type alias handling of TimeOnly & DateOnly has not been taken into account in this validation?

### Failing test

```csharp
#if NET6_0_OR_GREATER

        record RecordContainingDateAndtimeOnly {
            public DateOnly? NullableDate { get; set; }
            public TimeOnly? NullableTime { get; set; }
            public DateOnly Date { get; set; }
            public TimeOnly Time { get; set; }
        }

        [Fact]
        public async Task DateOnlyTimeOnly_Nullable_Serde() {

            var data = Enumerable.Range(0, 1_000).Select(i => new RecordContainingDateAndtimeOnly {
                NullableDate = i==0? null : DateOnly.MinValue.AddDays(i),
                NullableTime = i==0? null: TimeOnly.MinValue.AddMinutes(i),
                Date =  DateOnly.MinValue.AddDays(i+1),
                Time = TimeOnly.MinValue.AddMinutes(i+1),
            }).ToList();

            using var ms = new MemoryStream();
            await ParquetSerializer.SerializeAsync(data, ms);

            ms.Position = 0;
            IList<RecordContainingDateAndtimeOnly> data2 = await ParquetSerializer.DeserializeAsync<RecordContainingDateAndtimeOnly>(ms);

            Assert.Equivalent(data2, data);
        }

#endif
```
```

**Comment by aloneguid** [comment]
Thanks for providing with the reproducible test, appreciate it. I'll have a look asap.

**Comment by aloneguid** [comment]
This is fixed in 4.23.0, please feel free to validate and re-open if any issues.
